experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -665,6 +702,147 @@ func (r *ExpanderReconciler) enqueueAllFromGVK(ctx context.Context, _ client.Obj
 	return reqs
 }
 
+func (r *ExpanderReconciler) reconcileDelete(ctx context.Context, logger logr.Logger, inputcr unstructured.Unstructured) (ctrl.Result, error) {
+	logger = logger.WithName("Delete")
+	if !controllerutil.ContainsFinalizer(&inputcr, finalizerName) {
+		return ctrl.Result{}, nil
+	}
+	planNN := types.NamespacedName{
+		Name:      r.InputGVR.Resource + "-" + inputcr.GetName(),
+		Namespace: inputcr.GetNamespace(),
+	}
+	plancr := compositionv1alpha1.Plan{}
+	if err := r.Client.Get(ctx, planNN, &plancr); err != nil {
+		logger.Error(err, "Unable to fetch Plan Object", "plan", planNN)
+		return ctrl.Result{}, err
+	}
+	annotations := plancr.GetAnnotations()
+	planLabels := plancr.GetLabels()
+	stageList, ok := annotations[stagesAnnotation]
+	if !ok {
+		err := fmt.Errorf("Plan is missing stage order annotation")
+		logger.Error(err, "Unable to fetch stage order from Plan", "Plan", planNN)
+		return ctrl.Result{}, err
+	}
+	stages := strings.Split(stageList, ",")
+	numFound := 0
+	for i := len(stages) - 1; i >= 0; i-- {
+		r.Recorder.Eventf(&inputcr, corev1.EventTypeNormal, "Delete", "Deleting objects for stage %s", stages[i])
+		nsList, ok := annotations[nsAnnotation]
+		if !ok {
+			err := fmt.Errorf("Plan is missing Namespace annotation")
+			logger.Error(err, "Unable to fetch Namespaces from Plan", "Plan", planNN)
+			return ctrl.Result{}, err
+		}
+		namespaces := strings.Split(nsList, ",")
+		namespaces = append(namespaces, inputcr.GetNamespace())
```

Comment:
why is this necessary ?

Reply:
The facade namespace might have resources in it (as is the case in most of our tests). ApplySet adds a list of "additional" namespaces -- namely, namespaces other than the inputcr's namespace.